src/test/java/edu/hm/hafner/analysis/parser/PREfastParserTest.java

```diff
@@ -26,25 +27,29 @@
     @Test
     public void testParse() throws IOException {
         Issues results = new PREfastParser().parse(openFile());
-
-        assertEquals(11, results.size());
+        assertThat(results).hasSize(11);
 
         Iterator<Issue> iterator = results.iterator();
+        checkWarnings(iterator.next(), Priority.NORMAL, "28101", 102,
+                "The Drivers module has inferred that the current function is a DRIVER_INITIALIZE function:  This is informational only. No problem has been detected.", "sys.c", TYPE);
 
-        Issue annotation = iterator.next();
-        checkWarning(annotation,
-                102, "The Drivers module has inferred that the current function is a DRIVER_INITIALIZE function:  This is informational only. No problem has been detected.",
-                "sys.c", TYPE, "28101", Priority.NORMAL);
+        checkWarnings(iterator.next(), Priority.NORMAL, "6014", 116,
+                "(PFD)Leaking memory 'device'.", "sys.c", TYPE);
 
-        annotation = iterator.next();
-        checkWarning(annotation,
-                116, "(PFD)Leaking memory 'device'.",
-                "sys.c", TYPE, "6014", Priority.NORMAL);
+        checkWarnings(iterator.next(), Priority.NORMAL, "28155", 137,
+                "The function being assigned or passed should be a DRIVER_UNLOAD function:  Add the declaration 'DRIVER_UNLOAD OnUnload;' before the current first declaration of OnUnload.", "sys.c", TYPE);
+    }
 
-        annotation = iterator.next();
-        checkWarning(annotation,
-                137, "The function being assigned or passed should be a DRIVER_UNLOAD function:  Add the declaration 'DRIVER_UNLOAD OnUnload;' before the current first declaration of OnUnload.",
-                "sys.c", TYPE, "28155", Priority.NORMAL);
+    private void checkWarnings(Issue issue, Priority priority, String category, int lineStartAndEnd, String message, String fileName, String type) {
```

Comment:
Die Methode wieder inline machen: sonst geht genau der Effekt verloren, dass man nicht sieht welcher Parameter welchem Wert entspricht.

Reply:
Und dann alle assertions in einen softly packen.